wordvecspace/wvspace.py

```diff
@@ -231,33 +228,301 @@ def _nearest_sorting(self, d, k):
+               [ 2224,  5281,  3560,  4622, 11886]], dtype=uint32)
+	>>> wv.get_nearest(vecs, k=5, examine_k=500,
+	... combination=True, combination_method='set_intersect')
+	array([[  523,  5969, 24149,  9772,   486]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_intersect', weights=[0.1, 0.9])
+        array([[ 2224,  5281,  3560,  4622, 11886]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500, combination=True,
+        ... combination_method='set_intersect', weights=[0.1, 0.9], include_distances=True)[1]
+        array([[0.2795, 0.2814, 0.2819, 0.282 , 0.2821]], dtype=float32)
+	>>> wv.get_nearest(vecs, k=5, examine_k=500,
+	... combination=True, combination_method='set_union')
+	array([[11087, 11304, 14483,  4538, 10737]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_union', weights=[0.1, 0.9])
+        array([[13942,   503, 28413, 27280, 26180]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500,
+        ... combination=True, combination_method='set_union', weights=[0.9, -0.9])
+        array([[ 2822,  2765, 12515, 66318,  8591]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, examine_k=500, combination=True,
+        ... combination_method='set_union', weights=[0.1, 0.9], include_distances=True)[1]
+        array([[0.0321, 0.0321, 0.0322, 0.0322, 0.0322]], dtype=float32)
+	>>> wv.get_nearest(vecs, k=5, combination=True, combination_method='distance')
+	array([[  523,  5969, 24149,  9772,   486]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='distance', weights=[0.5, 0.3])
+        array([[  523,  5969,   486, 24149,  9772]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='distance', weights=[0.5, 0.3], include_distances=True)[1]
+        array([[0.2425, 0.2435, 0.2435, 0.2439, 0.2441]], dtype=float32)
+	>>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector')
+	array([[  523,  5969, 24149,  9772,   486]], dtype=uint32)
+	>>> wv.get_nearest(vecs, k=5, combination=True,
+	... combination_method='vector', weights=[0.8, 0.2])
+	array([[  486,   509,   523, 24149,  3389]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='vector', weights=[0.5, 0.3])
+        array([[  523,  5969,   486, 24149,  9772]], dtype=uint32)
+        >>> wv.get_nearest(vecs, k=5, combination=True,
+        ... combination_method='vector', weights=[0.5, 0.3], include_distances=True)[1]
+        array([[0.0006, 0.0023, 0.0023, 0.003 , 0.0034]], dtype=float32)
+        """
+
+        if not combination:
+            return self._get_brute(v_w_i, k, metric,
+                                   combination, include_distances)
+
+        if combination and combination_method == 'set_intersect':
```

Comment:
Checking for combination is not important if you use `elif`

Reply:
Here and in all the following if conditions

Reply:
Also for the sake of being safe, I think there should be an else condition.

Reply:
noted